name: test-check-fixme-remains

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
  test-check-fixme-remains:
    runs-on: ubuntu-latest
    steps:
      - name: checkout repository
        uses: actions/checkout@v4

      - name: run check fixme remains (case1, fixme remains)
        id: check-fixme-remains
        uses: yiegle/actions/check-draft-code-remains@main
        with:
          draft-code: 'FIXME:'
          path-to-check: './check-draft-code-remains/test/fixme-remains'
        continue-on-error: true

      - name: tests check fixme remains by comparing the result
        run: |
          if [ ${{ steps.check-fixme-remains.outcome }} == "success" ]; then
            echo "Failed: returns success, but exptected to return error. because there is a file containing FIXME:."
            exit 1
          else
            echo "Succecced: There is a file containing FIXME:, and the action returns error."
          fi

      - name: run check fixme remains (case2, no fixme remains)
        id: check-no-fixme-remains
        uses: yiegle/actions/check-draft-code-remains@main
        with:
          draft-code: 'FIXME:'
          path-to-check: './check-draft-code-remains/test/no-fixme-remains'